### DISCUSSION TITLE
Remove support for Deprecated DTD in 4.1

### DISCUSSION THREAD

**Discussion by mscuthbert** [w3c-cg/musicxml#481] [Ideas]
The old MusicXML DTD has been deprecated since MusicXML 4.0 in favor of MusicXML Schemas.  Schemas are richer in what they can support, tend to live in one file, and have been the future of XML for over ten years.

MusicXML, however, began as a DTD and thus has supported DTDs for a long time.

I'm the new editor of the MusicXML spec, and while I've been a big advocate and commenter on it for decades, I'm not an XML expert the way Michael Good was.  One of the concerns in working towards a 4.1 release is whether I can quickly get up to speed with both Schemas (which I'm pretty comfortable with) and also the older DTDs (which my tooling doesn't automatically produce so have to be generated by hand).

I'm wondering how much use of the DTDs there still is for verification, not documentation (I still find the DTDs sometimes easier to read than the XML Schemas, if only because they're broken up to digestible parts).  Since musicxml.org went off line (still being investigated) I've seen that some programs are still using the DTDs.  The question is whether people who are using the DTDs are also upgrading to the latest specs.  (A program that wants the musicxml 2.0 DTD is in a different boat than one validating on the 4.0 DTD).

Opening up for discussion.  My lean is towards removal for future versions, if only to prevent 2 more decades of deferred maintenance debt.

(First hinted at in https://github.com/w3c/mnx/discussions/291#discussioncomment-4653555)

**Comment by Jojo-Schmitz** [comment]
musixcxml.org being down is reported in #480 (just connecting the dots)

**Reply by DavidMWebber** [reply]
Please DO NOT remove the DTD option.

My software (https://www.mozart.co.uk) has been importing/exporting MusicXML for a long time now and the DTD approach has always been working well.  MusicXML import/export is  a feature which I update only occasionally and I have to do a lot of 'revision' of the spec each time I do it.   I appreciate that Schemas may offer more than DTDs, but removal of the DTD would be a major obstacle for me upgrading to use MusicXML 4.

**Reply by mscuthbert** [reply]
Thanks David -- you were just in the nick of time -- I was going to soon go with the "having heard no objection...." route, but I think I can instead work on getting a stable URL for the DTD for the next version.

**Reply by DavidMWebber** [reply]
Thanks for your reply - I really hope you can keep the DTD as an available option.
